src/Core/Domain/Cart/QueryResult/CartInformation/CartShipping.php

```diff
@@ -53,22 +53,30 @@ class CartShipping
      */
     private $deliveryOptions;
 
+    /**
+     * @var bool
+     */
+    private $isFreeShippingCartRuleAdded;
+
     /**
      * @param string $shippingPrice
      * @param bool $freeShipping
      * @param CartDeliveryOption[] $deliveryOptions
      * @param int|null $selectedCarrierId
+     * @param bool $hasFreeShippingCartRule
      */
     public function __construct(
         string $shippingPrice,
         bool $freeShipping,
         array $deliveryOptions,
-        ?int $selectedCarrierId
+        ?int $selectedCarrierId,
+        bool $hasFreeShippingCartRule
     ) {
         $this->shippingPrice = $shippingPrice;
         $this->freeShipping = $freeShipping;
         $this->deliveryOptions = $deliveryOptions;
         $this->selectedCarrierId = $selectedCarrierId;
+        $this->isFreeShippingCartRuleAdded = $hasFreeShippingCartRule;
     }
```

Comment:
The property name is `isFreeShippingCartRuleAdded` and the injected variable is called `$hasFreeShippingCartRule` .

That's a minor thing but is it possible to give it the same name ? (I'm not blocking this PR for this though)

Reply:
Yes, it is possible. It happened because .js trims  the prefix of object property. That's why I though it must start with 'is' but also proved to be correct with 'has'. Big thanks! :slightly_smiling_face: